dev-tools/reconfigurator-exec/src/main.rs

```diff
@@ -0,0 +1,175 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Execute blueprints from the command line
+
+use anyhow::Context;
+use anyhow::anyhow;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db;
+use nexus_db_queries::db::DataStore;
+use nexus_reconfigurator_execution::realize_blueprint;
+use nexus_types::deployment::UnstableReconfiguratorState;
+use omicron_uuid_kinds::BlueprintUuid;
+use omicron_uuid_kinds::OmicronZoneUuid;
+use slog::info;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use update_engine::EventBuffer;
+use update_engine::NestedError;
+use update_engine::display::LineDisplay;
+use update_engine::display::LineDisplayStyles;
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let args = ReconfiguratorExec::parse();
+
+    if let Err(error) = args.exec().await {
+        eprintln!("error: {:#}", error);
+        std::process::exit(1);
+    }
+
+    Ok(())
+}
+
+/// Execute blueprints from the command line
+#[derive(Debug, Parser)]
+struct ReconfiguratorExec {
+    /// log level filter
+    #[arg(
+        env,
+        long,
+        value_parser = parse_dropshot_log_level,
+        default_value = "info",
+        global = true,
+    )]
+    log_level: dropshot::ConfigLoggingLevel,
+
+    dns_server: SocketAddr,
```

Comment:
I think I'll change this to default to one of the three internal DNS servers, which have the same underlay IPs in all real deployments.  The only case this would be wrong is in a simulated deployment (like `cargo xtask omicron-dev run-all`).  It seems unlikely this would be used there but if so you could always override this.

Reply:
Done.